letsencrypt/acme/other.py

```diff
@@ -0,0 +1,86 @@
+"""JSON objects in ACME protocol other than messages."""
+import logging
+
+from Crypto import Random
+import Crypto.Hash.SHA256
+import Crypto.Signature.PKCS1_v1_5
+
+from letsencrypt.acme import jose
+from letsencrypt.acme import util
+
+
+class Signature(util.JSONDeSerializable, util.ImmutableMap):
+    """ACME signature.
+
+    :ivar str alg: Signature algorithm.
+    :ivar str sig: Signature.
+    :ivar str nonce: Nonce.
+
+    :ivar jwk: JWK.
+    :type jwk: :class:`letsencrypt.acme.jose.JWK`
+
+    .. todo:: Currently works for RSA keys only.
+
+    """
+    __slots__ = ('alg', 'sig', 'nonce', 'jwk')
+    schema = util.load_schema('signature')
+
+    NONCE_LEN = 16
+    """Size of nonce in bytes, as specified in the ACME protocol."""
+
+    @classmethod
+    def from_msg(cls, msg, key, nonce=None):
+        """Create signature with nonce prepended to the message.
+
+        .. todo:: Change this over to M2Crypto... PKey
```

Comment:
I have tried to kill this todo in several PRs now and it keeps on coming back.  Would you mind removing it in your branch? I think we are going to try to stay away from M2Crypto.